tests/frame_manager_test.go

```diff
@@ -41,32 +41,18 @@ func TestWaitForFrameNavigationWithinDocument(t *testing.T) {
 
 					resp := p.Goto(tb.staticURL("/nav_in_doc.html"), tb.toGojaValue(&common.FrameGotoOptions{
 						WaitUntil: common.LifecycleEventNetworkIdle,
-						Timeout:   timeout, // interpreted as ms
+						Timeout:   time.Duration(timeout.Milliseconds()), // interpreted as ms
 					}))
 					require.NotNil(t, resp)
-
-					// Callbacks that are initiated internally by click and WaitForNavigation
-					// need to be called from the event loop itself, otherwise the callback
-					// doesn't work. The await below needs to first return before the callback
-					// will resolve/reject.
-					var wfnPromise, cPromise *goja.Promise
-					err := tb.await(func() error {
-						wfnPromise = p.WaitForNavigation(tb.toGojaValue(&common.FrameWaitForNavigationOptions{
-							Timeout: timeout, // interpreted as ms
-						}))
-						cPromise = p.Click(tc.selector, nil)
-
-						assert.Equal(t, goja.PromiseStatePending, wfnPromise.State())
-						assert.Equal(t, goja.PromiseStatePending, cPromise.State())
-
-						return nil
-					})
-					if err != nil {
-						return err
-					}
-
-					assert.Equal(t, goja.PromiseStateFulfilled, wfnPromise.State())
-					assert.Equal(t, goja.PromiseStateFulfilled, cPromise.State())
+					wfnPromise := p.WaitForNavigation(tb.toGojaValue(&common.FrameWaitForNavigationOptions{
+						Timeout: time.Duration(timeout.Milliseconds()), // interpreted as ms
+					}))
+					cPromise := p.Click(tc.selector, nil)
+					tb.promiseThen(tb.promiseAll(wfnPromise, cPromise), func(_ goja.Value) {
+						// this is a bit pointless :shrug:
```

Comment:
Why is it pointless?

Reply:
Because we are basically checkking that they are fulfilled in the callback that gets called when they get fulfilled - in practice not checking anything.

Reply:
Hmm, should we remove the checks in closure then?

Reply:
:shrug: then it will be empty and a bit ... strange. I guess I can make certain that it gets called by setting a boolean :shrug:

Reply:
This test is a bit pointless to begin with. The only reason it was initially like this is because `WaitForNavigation()` panicked on same-document navigations, and this tested that it didn't panic anymore.

But to make it useful, we should make it more functional, and assert that e.g. a certain element is visible after navigating on the same page.

Doesn't have to be done here, just FYI.